lib/redux/actions/async.js

```diff
@@ -140,8 +140,12 @@ export function doLogin(creds, opts) {
         memberships: results[2]
       }));
 
-      // detect if a VCA here and redirect to clinic user select screen
+      // detect if a DSA here and redirect to data storage screen
+      if (!personUtils.userHasDSA(results[0].user, results[1], results[2])) {
```

Comment:
I think the approach here could be simplified considerably by getting rid of the utility function and doing a `getState` check. Since `loginSuccess` is already dispatched, I believe the state should be updated with the results, and then you can just check for the length of the `targetUsersForUpload` array. Perhaps give that a try and see if it covers all the same cases. I think it will. If you think about it, we're already doing the kind of logic that is in that utility function in our reducers when we update the state based on the results of logging in...